pkgs/applications/editors/vscode/extensions/remote-ssh/default.nix

```diff
@@ -9,29 +9,61 @@
 let
   inherit (vscode-utils) buildVscodeMarketplaceExtension;
 
-  # patch runs on remote machine hence use of which
-  # links to local node if version is 12
+  # As VS Code executes this code on the remote machine
+  # we test to see if we can build Node 14 from Nixpkgs
+  # otherwise we use the local Node if it is version 14
   patch = ''
     f="$HOME/.vscode-server/bin/$COMMIT_ID/node"
-    localNodePath=''$(which node)
-    if [ -x "''$localNodePath" ]; then
-      localNodeVersion=''$(node -v)
-      if [ "\''${localNodeVersion:1:2}" = "12" ]; then
-        echo PATCH: replacing ''$f with ''$localNodePath
-        rm ''$f
-        ln -s ''$localNodePath ''$f
+    echo "VS Code Node: $f"
+
+    # Check if VS Code Server has a non-working Node or the wrong version of Node
+    if ! nodeVersion=$($f -v) || [ "\''${nodeVersion:1:2}" != "14" ]; then
+      echo "VS Code Node Version: $nodeVersion"
+
+      nodePath=$(nix-build "<nixpkgs>" -A nodejs-14_x --no-out-link)/bin/node
```

Comment:
Won't that be garbage collected?

Reply:
the code runs on every connect, so if it stops working for whatever reason (like Node being gc'd) it'll just build Node again

Reply:
Could we easily prevent it from being garbage collected like creating an out link?

Reply:
Fixed :+1: